tests/tests.py

```diff
@@ -0,0 +1,223 @@
+    """
+    Tests functions in latin_text_syllabification.
+    """
+
+    def test_syllabify_word(self):
+        """Tests syllabify_word."""
+        # Read test words from csv file and get syllable boundaries.
+        # ie. "Be-ne-dic-tus" -> [2, 4, 7]
+        test_words = {}
+        with open(
+            "tests/word_syllabification_tests.csv", "r", encoding="utf-8-sig"
+        ) as f:
+            reader = csv.reader(f)
+            for row in reader:
+                word = row[0]
+                syl_list = row[1].split("-")
+                syl_bound = 0
+                syl_bounds = []
+                for syl in syl_list[:-1]:
+                    syl_bound += len(syl)
+                    syl_bounds.append(syl_bound)
+                test_words[word] = syl_bounds
+        for word, expected in test_words.items():
+            with self.subTest(word=word):
+                self.assertEqual(syllabify_word(word, return_string=False), expected)
+
+    def test_split_word_by_syl_bounds(self):
+        """
+        Tests split_word_by_syl_bounds.
+
+        Test 1, 2, and 2+ syllable words.
+        """
+        test_words = {"Benedictus": "Be-ne-dic-tus", "qui": "qui", "venit": "ve-nit"}
+        word_syl_bounds = {"Benedictus": [2, 4, 7], "qui": [], "venit": [2]}
+        for word, expected in test_words.items():
+            with self.subTest(word=word):
+                self.assertEqual(
+                    "".join(split_word_by_syl_bounds(word, word_syl_bounds[word])),
+                    expected,
+                )
+
+
+class TestCantusTextSyllabification(unittest.TestCase):
+    """
+    Tests functions in cantus_text_syllabification.
+    """
+
+    def test_clean_test(self):
+        """Tests _clean_text."""
+        initial_text = "abcdefg @#$&*[^@]#${}|~[]/|\\"
+        expected_text = "abcdefg #[]#{}|~[]|"
+        self.assertEqual(_clean_text(initial_text), expected_text)
+
+    def test_prepare_string_for_syllabification(self):
+        """Tests _prepare_string_for_syllabification."""
+        str_hyphen_start = "-ABCDEFG"
+        str_hyphen_end = "ABCDEFG-"
+        self.assertEqual(
+            _prepare_string_for_syllabification(str_hyphen_start),
+            ("ABCDEFG", True, False),
+        )
+        self.assertEqual(
+            _prepare_string_for_syllabification(str_hyphen_end),
+            ("ABCDEFG", False, True),
+        )
+
+    def test_split_text_sections(self):
+        """
+        Tests _split_text_sections.
+
+        Note: Sections are defined by pipes and curly braces.
+        """
+        start_str = "Benedictus | ~qui [venit in] | {nomine Domini} amen"
+        sectioned = [
+            "Benedictus",
+            "|",
+            "~qui [venit in]",
+            "|",
+            "{nomine Domini}",
+            "amen",
+        ]
+        self.assertEqual(_split_text_sections(start_str), sectioned)
+
+    def test_stringify_syllabified_text(self):
```

Comment:
Since these functions will be handling manually-syllabified text from CantusDB, do we want a wider range of test cases and expected results? (At the very least, should we test one or two words other than "sanctus"?) How robust is `stringify_syllabified_text` to weird, human-error sort of situations? (Again, this might become clear as I look through more of the code.)

Reply:
Good question...looking again at that function, I think what would trip it up is if what was passed to the function was either (a) not in the expected nested list structure or (b) contained items that could not be cast to strings. (Are there others?) 

I can't quite see how someone could enter something like that into a text form, but maybe we can put some sort of error handling in the function? Let me think about this.

Reply:
Thinking about it a bit more, only syllabified volpiano should ever end up in this function - if the original volpiano was not well-formed, a failure would have come up while attempting to syllabify it. So probably no changes needed here.